### PR TITLE
 #299 mock pythesint in all tests

### PR DESCRIPTION
**Pull request by akorosov**
@korvinos , can you please have a look at this PR?

### CODE REVIEW

File: nansat/tests/nansat_test_base.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#               http://www.gnu.org/licenses/gpl-3.0.html
+# ------------------------------------------------------------------------------
+from __future__ import print_function, absolute_import, division
+import os
+import sys
+import unittest
+import tempfile
+import pythesint
+
+if sys.version_info.major == 2:
```
Comment (korvinos) [inline nansat/tests/nansat_test_base.py]: I thought you got rid of that in some PR in the past. I remember that I made a review

File: nansat/tests/nansat_test_base.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    from mock import patch, PropertyMock, Mock, MagicMock, DEFAULT
+else:
+    from unittest.mock import patch, PropertyMock, Mock, MagicMock, DEFAULT
+
+from nansat.tests import nansat_test_data as ntd
+
+class NansatTestBase(unittest.TestCase):
+    def setUp(self):
```
Comment (korvinos) [inline nansat/tests/nansat_test_base.py]: A blank line before the method

File: nansat/tests/nansat_test_base.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+import tempfile
+import pythesint
+
+if sys.version_info.major == 2:
+    from mock import patch, PropertyMock, Mock, MagicMock, DEFAULT
+else:
+    from unittest.mock import patch, PropertyMock, Mock, MagicMock, DEFAULT
+
+from nansat.tests import nansat_test_data as ntd
+
```
Comment (korvinos) [inline nansat/tests/nansat_test_base.py]: 2 blank lines before the class

File: nansat/tests/nansat_test_base.py [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+#
+# Author:       Anton Korosov
+#
+# Created:      20.03.2018
+# Copyright:    (c) NERSC
+# Licence:      This file is part of NANSAT. You can redistribute it or modify
+#               under the terms of GNU General Public License, v.3
+#               http://www.gnu.org/licenses/gpl-3.0.html
+# ------------------------------------------------------------------------------
+from __future__ import print_function, absolute_import, division
```
Comment (korvinos) [inline nansat/tests/nansat_test_base.py]: I remember a bug when you told me that print_function does not work.

File: nansat/tests/nansat_test_base.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+        self.mock_pti['get_gcmd_instrument'].return_value=dict(short_name='MODIS')
+        self.mock_pti['get_gcmd_platform'].return_value=dict(short_name='AQUA')
+        self.mock_pti['get_wkv_variable'].return_value=dict(short_name='swathmask')
+
+    def tearDown(self):
+        self.patcher.stop()
+        try:
+            os.unlink(self.tmpfilename)
+        except OSError:
+            pass
```
Comment (korvinos) [inline nansat/tests/nansat_test_base.py]: A blank line in the very end of a file

File: nansat/tests/nansat_test_base.py [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+
+class NansatTestBase(unittest.TestCase):
+    def setUp(self):
+        self.test_file_gcps = os.path.join(ntd.test_data_path, 'gcps.tif')
+        self.test_file_stere = os.path.join(ntd.test_data_path, 'stere.tif')
+        self.test_file_complex = os.path.join(ntd.test_data_path, 'complex.nc')
+        self.test_file_arctic = os.path.join(ntd.test_data_path, 'arctic.nc')
+        self.tmp_data_path = ntd.tmp_data_path
+        self.default_mapper = 'generic'
+        fd, self.tmpfilename = tempfile.mkstemp(suffix='.nc', dir=ntd.tmp_data_path)
```
Comment (korvinos) [inline nansat/tests/nansat_test_base.py]: I would make tmp_filename

File: nansat/tests/nansat_test_base.py [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                                                  get_gcmd_platform=DEFAULT)
+        self.mock_pti = self.patcher.start()
+        self.mock_pti['get_gcmd_instrument'].return_value=dict(short_name='MODIS')
+        self.mock_pti['get_gcmd_platform'].return_value=dict(short_name='AQUA')
+        self.mock_pti['get_wkv_variable'].return_value=dict(short_name='swathmask')
+
+    def tearDown(self):
+        self.patcher.stop()
+        try:
+            os.unlink(self.tmpfilename)
```
Comment (korvinos) [inline nansat/tests/nansat_test_base.py]: I would make tmp_filename

**Comment by akorosov** [comment]
@korvinos, I've fixed everything. Can you give a final check and approve?

**Pull request merged**